### PR TITLE
propagate new bc::settings class

### PR DESCRIPTION
**Pull request by toxeus**
I'm planning to extend the parser once all constants have been moved into `bc::settings`.

### CODE REVIEW

File: src/sessions/session_inbound.cpp [hunk 1]
```diff
@@ -35,8 +35,9 @@ using namespace bc::message;
 using namespace bc::network;
 using namespace std::placeholders;
 
-session_inbound::session_inbound(full_node& network, safe_chain& chain)
-  : session<network::session_inbound>(network, true),
+session_inbound::session_inbound(full_node& network, safe_chain& chain,
+    const bc::settings& bitcoin_settings)
```
Comment (evoskuil) [inline src/sessions/session_inbound.cpp]: Node exposes settings so passing settings is redundant.

File: src/protocols/protocol_block_sync.cpp [hunk 2]
```diff
@@ -60,7 +61,7 @@ void protocol_block_sync::start()
     protocol_timer::start(monitor_interval, BIND1(handle_event, _1));
 
     chain_.subscribe_headers(BIND4(handle_reindexed, _1, _2, _3, _4));
-    SUBSCRIBE2(block, handle_receive_block, _1, _2);
+    SUBSCRIBE3(block, handle_receive_block, bitcoin_settings_, _1, _2);
```
Comment (evoskuil) [inline src/protocols/protocol_block_sync.cpp]: It seems that protocols should be constructed with settings, not passed on each handler invoke.

Comment (toxeus) [reply src/protocols/protocol_block_sync.cpp]: Agreed. Fixed.

File: src/sessions/session_inbound.cpp (hunk 1 above)
Comment (toxeus) [reply src/sessions/session_inbound.cpp]: Good catch! Done.

**Pull request merged**